lib/services/flowdock.ts

```diff
@@ -15,129 +15,137 @@ limitations under the License.
 */
 
 import * as Promise from 'bluebird';
-import {
-    Session
-} from 'flowdock';
+import { Session } from 'flowdock';
 import * as _ from 'lodash';
 import * as path from 'path';
 import * as request from 'request-promise';
 import {
-    FlowdockHandle,
-    FlowdockMessage,
-    FlowdockMessageEmitContext,
-} from '../services/flowdock-types';
-import {
-    MessageEmitResponse,
-    MessageEvent,
-    MessageWorkerEvent,
-    ReceiptContext,
+    DataHub, MessengerEmitResponse, MessengerEvent, ReceiptContext, TransmitContext
 } from '../utils/message-types';
-import {
-    MessageService
-} from './message-service';
-import {
-    ServiceEmitter,
-    ServiceListener,
-} from './service-types';
+import { FlowdockEmitContext, FlowdockHandle, FlowdockMessage } from './flowdock-types';
+import { MessageService } from './message-service';
+import { ServiceEmitter, ServiceListener } from './service-types';
 
-/**
- * Class for interacting with the Discourse API
- * Is a MessageService, ServiceListener and ServiceEmitter
- */
-export class FlowdockService extends MessageService implements ServiceEmitter, ServiceListener {
-    private static session = new Session(process.env.FLOWDOCK_LISTENER_ACCOUNT_API_TOKEN);
+export class FlowdockService extends MessageService implements ServiceEmitter, ServiceListener, DataHub {
     private static _serviceName = path.basename(__filename.split('.')[0]);
-    private flowIdToFlowName = new Map<string, string>();
+    private static session = new Session(process.env.FLOWDOCK_LISTENER_ACCOUNT_API_TOKEN);
```

Comment:
This from @hedss, applies more generally, but I'm putting it here to start a thread in the review layout.
As a general comment, I'm not crazy about the use of static instances of the SDKs used in the code. The Github service creates a new instance of the Github API for each ServiceListener/Emitter it provides. Static implementation provides no ability for multiple bots running in the same deployment to service different orgs/users, or allow for failure in case the static API instance locks up.
Resource is cheap.